app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt

```diff
@@ -1124,7 +1124,12 @@ class ComposeActivity :
     private fun handleCloseButton() {
         val contentText = binding.composeEditField.text.toString()
         val contentWarning = binding.composeContentWarningField.text.toString()
-        if (viewModel.didChange(contentText, contentWarning)) {
+        if (viewModel.composeKind == ComposeKind.NEW && viewModel.isEmpty(contentText, contentWarning)) {
```

Comment:
This sort of business logic probably belongs in the view model, probably in the existing `didChange` function (perhaps renamed).

The function could then return an enum that corresponds to the type of confirmation dialog the activity should show (if any).

Something like:

```kotlin
when (viewModel.didChange(contextText, contentWarning)) {
    NO_CHANGE -> { /* Stop uploads and finish */ }
    SAVE_OR_DISCARD -> getSaveAsDraftOrDiscardDialog(...)
    UPDATE_OR_DISCARD -> getUpdateDraftOrDiscardDialog(...)
    CONTINUE_OR_DISCARD -> getContinueEditingOrDiscardDialog(...)
}
```

Reply:
Refactored and moved the logic to the view model.